probe_scraper/parsers/repositories.py

```diff
@@ -154,19 +155,26 @@ def parse_v2(self, filename=None) -> dict:
                 )
                 listing = remove_none(listing)
                 model_validation.validate_as(listing, "Application")
-                listings.append(listing)
+                app_listings.append(listing)
+
         return {
             "libraries": repos["libraries"],
-            "applications": listings,
+            "applications": app_listings,
         }
 
     def _v2_to_v1(self, filename):
         repos_v2 = self.parse_v2(filename)
         repos = {}
         for lib in repos_v2["libraries"]:
-            v1_name = lib.pop("v1_name")
-            lib["app_id"] = v1_name
-            repos[v1_name] = lib
+            variants = lib.pop("variants")
+            for variant in variants:
+                newlib = {**lib, **variant}
+                newlib.pop("library_name")
+                newlib["library_names"] = [newlib["dependency_name"]]
+                newlib.pop("dependency_name")
+                v1_name = newlib.pop("v1_name")
+                newlib["app_id"] = v1_name
+                repos[v1_name] = newlib
```

Comment:
I don't think we have any coding styles I can fall back on here, but something about this felt like it was hard to parse. The following may be easier:
```suggestion
                lib_info = {**lib, **variant}
                v1_name = lib_info["v1_name"]
                lib_info["library_names"] = [lib_info["dependency_name"]]
                lib_info["app_id"] = v1_name
                
                del lib_info["library_name"]
                del lib_info["dependency_name"]
                del lib_info["v1_name"]
                
                repos[v1_name] = lib_info
```

Reply:
Ah, the rest of this file uses `pop`. Feel free to disregard

Reply:
Your way looks better, I agree. May as well take it.